src/main/java/appeng/items/contents/NetworkToolViewer.java

```diff
@@ -41,94 +43,28 @@ public NetworkToolViewer( final ItemStack is, final IGridHost gHost )
 	{
 		this.is = is;
 		this.gh = gHost;
-		this.inv = new AppEngInternalInventory( null, 9 );
+		this.inv = new AppEngInternalInventory( this, 9 );
+		this.inv.setFilter( new NetworkToolInventoryFilter() );
 		if( is.hasTagCompound() ) // prevent crash when opening network status screen.
 		{
 			this.inv.readFromNBT( Platform.openNbtData( is ), "inv" );
 		}
 	}
 
 	@Override
-	public int getSizeInventory()
+	public void saveChanges()
 	{
-		return this.inv.getSizeInventory();
+		// TODO: should not be needed
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)